src/views/modal_handler.rs

```diff
@@ -113,50 +143,90 @@ impl DisplayModal {
     }
 
     pub fn view(&self) -> Element<Message> {
-        let mut button_row = Row::new();
-        let mut text_style = TextStyle {
-            text_color: Color::new(0.447, 0.624, 0.812, 1.0),
-        };
-
-        if self.is_warning {
-            text_style = TextStyle {
-                text_color: Color::new(0.988, 0.686, 0.243, 1.0),
-            };
-            button_row = button_row.push(
-                button("Exit without saving")
-                    .on_press(Message::ModalHandle(ModalMessage::ExitApp))
-                    .style(MODAL_CANCEL_BUTTON_STYLE.get_button_style()),
-            ); // Exits the application
-            button_row = button_row
-                .push(
-                    button("Return to app")
+        match &self.modal_type {
+            Some(ModalType::Warning {
+                title,
+                body,
+                load_config,
+            }) => {
+                let mut button_row = Row::new();
+                let text_style = TextStyle {
+                    text_color: Color::new(0.988, 0.686, 0.243, 1.0),
+                };
+
+                if *load_config {
+                    button_row = button_row
+                        .push(
+                            button("Continue and load a new config")
+                                .on_press(Message::ModalHandle(ModalMessage::LoadFile))
+                                .style(MODAL_CANCEL_BUTTON_STYLE.get_button_style()),
+                        )
+                        .push(
+                            button("Return to app")
+                                .on_press(Message::ModalHandle(ModalMessage::HideModal))
+                                .style(MODAL_CONNECT_BUTTON_STYLE.get_button_style()),
+                        )
+                        .spacing(120);
+                } else {
+                    button_row = button_row
+                        .push(
+                            button("Exit without saving")
+                                .on_press(Message::ModalHandle(ModalMessage::ExitApp))
+                                .style(MODAL_CANCEL_BUTTON_STYLE.get_button_style()),
+                        )
+                        .push(
+                            button("Return to app")
+                                .on_press(Message::ModalHandle(ModalMessage::HideModal))
+                                .style(MODAL_CONNECT_BUTTON_STYLE.get_button_style()),
+                        )
+                        .spacing(220);
+                }
+
+                container(
```

Comment:
The container code repeats for different dialogs too.

I think we should try and DRY up the code, and make the file as short as possible, with minimal repetition...

Reply:
You could maybe put some of those strings into the different entries in the enum (text for each button, etc) so the code could then just pull it out into variables in a match and then layout the view using those variables once...

let (b1_text, b2_text, etc) = match dialog {
   HardwareDetails(x, y) => (x, y, "njkds"),
   other dialogs
};

                    button_row = button_row
                        .push(
                            button(b1_text)
                                .on_press(Message::ModalHandle(ModalMessage::LoadFile))
                                .style(MODAL_CANCEL_BUTTON_STYLE.get_button_style()),
                        )
                        .push(
                            button(b2_text)
                                .on_press(Message::ModalHandle(ModalMessage::HideModal))
                                .style(MODAL_CONNECT_BUTTON_STYLE.get_button_style()),
                        )
                        .spacing(120);

the on_press might be harder to do, maybe need to return a closure in the match...